### ISSUE TITLE
Broken vector functionality

### ISSUE THREAD

**Issue by maleadt** [JuliaGPU/GPUArrays.jl#184]
https://github.com/JuliaGPU/GPUArrays.jl/blob/3cfc810c74add5900fd08e273473afb97840d52f/src/vectors.jl#L8
No buffer in there.

**Comment by vchuravy** [comment]
Yeah just came across this the whole vector testsuite is not being run.

We wanted to do `push!(CuArray([1]), 1)`.
